### PR TITLE
CMake test: storage_cuda without CUDA compiler

### PR DESCRIPTION
**Pull request by havogt**
Adds infrastructure for testing CMake projects using GridTools:
- Put a CMake project in `test/cmake/<testname>`.
- Add `add_cmake_test(<testname>)`.
- The CMake project should produce an executable called main which exits with exit code 0 on success. Later, this can be refined as needed.
- The configure & build & run will run as part of ctest with lable `cmake`.

### CODE REVIEW

**Comment by havogt** [comment]
launch strgrid

File: tests/cmake/storage_cuda/main.cpp [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#include <gridtools/storage/builder.hpp>
+#include <gridtools/storage/cuda.hpp>
+
+int main() {
+    auto builder = gridtools::storage::builder<gridtools::storage::cuda>.type<int>().dimensions(2);
+    auto my_storage = builder();
+    my_storage->host_view()(0) = 10;
+    my_storage->get_target_ptr();
```
Comment (havogt) [inline tests/cmake/storage_cuda/main.cpp]: This is enough to test CUDA runtime is working, right? It does a sync?!

Comment (anstaf) [reply tests/cmake/storage_cuda/main.cpp]: Strictly speaking this is not enough.
The proper test should take `get_target_ptr()` pointer; copy the memory back to the host using `cudaMemcpy`; cast to `int` and check that it is `10`

Comment (havogt) [reply tests/cmake/storage_cuda/main.cpp]: Agree. I'll add it.

**Comment by havogt** [comment]
launch jenkins

File: tests/cmake/storage_cuda/main.cpp (hunk 1 above)
Comment (havogt) [reply tests/cmake/storage_cuda/main.cpp]: done

**Pull request merged**